test/cmd/apply.sh

```diff
@@ -444,6 +444,9 @@ run_kubectl_server_side_apply_tests() {
 }
 __EOF__
 
+  # Ensure the CRD has been created
```

Comment:
This is not the correct check... the CRD will always be created here

The issue is whether the API server has recognized and started serving the associated CR API yet

Reply:
@liggitt Is it possible to do a wait on a Condition for a CRD? (e.g. `kubectl wait --for=condition=Ready mygroup.example.com/myobj --timeout=30s`)

Reply:
I think we want to wait until kubectl observes the expected API available in discovery (`kubectl api-resources` contains the new API)

Reply:
@seans3 you could wait for condition `Established`, although I am not certain that guarantees that it is being served https://github.com/kubernetes/kubernetes/blob/2585d78c79054a6d85620eb5430a1498bb747967/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go#L279